noggin/tests/unit/form/test_base.py

```diff
@@ -51,3 +51,30 @@ class TestForm(ModestForm):
     ):
         form = TestForm()
         assert form.is_submitted()
+
+
+def test_csvlistfield(client):
+    class DummyForm(FlaskForm):
+        csvfield = CSVListField('csv field')
+
+    form = DummyForm()
+
+    # check normal use
+    form.csvfield.process_formdata(["one,two,three"])
+    assert form.csvfield.data == ["one", "two", "three"]
+
+    # check we strip single spaces string, and return []
+    form.csvfield.process_formdata([" "])
+    assert form.csvfield.data == []
```

Comment:
You could also use [parametrize](https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) to generate test functions independant from each other.